### PR TITLE
Update dois java implementation

### PR DESCRIPTION
**Pull request by cookersjs**
This is a modification to the Update DOIs script that clones data-release-pipeline and runs the new version of the script. Some config files in the form of a properties (necessary) and report (optional) are now required.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,9 @@ tmp/
 news.html
 inference_stats.png
 website/html/img-tmp
+Secrets.pm
+release-config.pm
```
Comment (SolomonShorser-OICR) [inline .gitignore]: What is this release-config.pm file?

File: scripts/release/update_dois/setup_update_dois.pl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# Log::Log4perl->init(\$LOG_CONF);
+my $logger = get_logger(__PACKAGE__);
+
+my $arg;
+if ($ARGV[0] && $ARGV[0] eq '-clone') {
+  $arg = $ARGV[0];
+}
+
+if (-e "data-release-pipeline" || !$arg) {
+  chdir "data-release-pipeline/update-dois";
```
Comment (SolomonShorser-OICR) [inline scripts/release/update_dois/setup_update_dois.pl]: Maybe a good idea to do a `git pull` to get the latest changes, when the directory already exists.

File: .gitignore (hunk 1 above)
Comment (cookersjs) [reply .gitignore]: Its the GKB::Release::Config file. If you look at the release-container branch, and in the run_release_container.sh script: 

https://github.com/reactome/Release/blob/release-container/container/run_release_container.sh#L65

It was a config file that just needed to be renamed

File: scripts/release/update_dois/setup_update_dois.pl (hunk 2 above)
Comment (cookersjs) [reply scripts/release/update_dois/setup_update_dois.pl]: Changed

**Comment by cookersjs** [comment]
@SolomonShorser-OICR Changes have been made to UpdateDOIs

File: modules/GKB/Release/Steps/UpdateDOIs.pm [hunk 3]
```diff
@@ -25,17 +30,19 @@ override 'run_commands' => sub {
...
             ["mysqldump -u$user -p$pass -h$gkcentral_host --lock-tables=FALSE $gkcentral > $gkcentral.dump"]
         ]
     );
-    
-    my $live_run = $self->user_input->{'live_run'}->{'response'} =~ /^y/i ? '-live_run' : '';
-    my @args = ("-user", $user, "-pass", $pass, "-release_db", $db, "-release_db_host", $host, "-curator_db", $gkcentral, "-curator_db_host", $gkcentral_host, $live_run);
-    $self->cmd("Running script to update DOIs for $db and $gkcentral",[["perl update_dois.pl @args > update_dois.out 2>> update_dois.err"]]);
+
+    if ($self->user_input->{'props'}->{'response'} !~ /^y/i) {
+      die "Please update the databaseTR, and authorID fields in update_dois/config.properties before running.\n";
```
Comment (SolomonShorser-OICR) [inline modules/GKB/Release/Steps/UpdateDOIs.pm]: What is "databaseTR"? Are there some simple instructions you could put into the script as comments?

Comment (cookersjs) [reply modules/GKB/Release/Steps/UpdateDOIs.pm]: TR stands for Test Reactome, while GK indicates the GK_Central database. I'll add those comments

Comment (SolomonShorser-OICR) [reply modules/GKB/Release/Steps/UpdateDOIs.pm]: Ok, the "TR" short-form wasn't clear to me.

**Pull request merged**